### PR TITLE
Analyzer - improve hiding secret arguments.

### PR DESCRIPTION
**Pull request by yakov-olkhovskiy**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

continuation of #60230

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit ab6ab8bfac4c0b5f2b941b678591cc7d318d6bdf with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60386/ab6ab8bfac4c0b5f2b941b678591cc7d318d6bdf/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60386/ab6ab8bfac4c0b5f2b941b678591cc7d318d6bdf/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Analyzer/Passes/QueryAnalysisPass.cpp [hunk 1]
```diff
@@ -5111,22 +5112,17 @@ ProjectionNames QueryAnalyzer::resolveFunction(QueryTreeNodePtr & node, Identifi
...
     bool allow_table_expressions = is_special_function_in;
     auto arguments_projection_names = resolveExpressionNodeList(function_node_ptr->getArgumentsNode(),
         scope,
         true /*allow_lambda_expression*/,
-        allow_table_expressions /*allow_table_expression*/);
+        allow_table_expressions /*allow_table_expression*/,
+        {secret_arguments.start, secret_arguments.count});
+
+    for (size_t n = secret_arguments.start; n < secret_arguments.start + secret_arguments.count; ++n)
+        arguments_projection_names[n] = "[HIDDEN]";
```
Comment (novikd) [inline src/Analyzer/Passes/QueryAnalysisPass.cpp:5124]: Let's better call `setMasked` here. Just take `n`th argument, convert it to `ConstantNode` and mark as a secret.

File: src/Analyzer/Passes/QueryAnalysisPass.cpp [hunk 2]
```diff
@@ -6111,7 +6107,7 @@ ProjectionNames QueryAnalyzer::resolveExpressionNode(QueryTreeNodePtr & node, Id
   * Example: CREATE TABLE test_table (id UInt64, value UInt64) ENGINE=TinyLog; SELECT plus(*) FROM test_table;
   * Example: SELECT *** FROM system.one;
   */
-ProjectionNames QueryAnalyzer::resolveExpressionNodeList(QueryTreeNodePtr & node_list, IdentifierResolveScope & scope, bool allow_lambda_expression, bool allow_table_expression)
+ProjectionNames QueryAnalyzer::resolveExpressionNodeList(QueryTreeNodePtr & node_list, IdentifierResolveScope & scope, bool allow_lambda_expression, bool allow_table_expression, const std::pair<size_t, size_t> & secrets)
```
Comment (novikd) [inline src/Analyzer/Passes/QueryAnalysisPass.cpp:6110]: So this change wouldn't be required.

File: src/Analyzer/IQueryTreeNode.h [hunk 3]
```diff
@@ -251,6 +251,11 @@ class IQueryTreeNode : public TypePromotion<IQueryTreeNode>
         return children;
     }
 
+    void setMasked(bool masked = true)
```
Comment (novikd) [inline src/Analyzer/IQueryTreeNode.h:254]: Let's add this method only to `ConstantNode`

**Review by novikd — APPROVED** [review]
LGTM

**Comment by yakov-olkhovskiy** [comment]
we need to unify it somehow I guess... I'll take a look

**Pull request merged**